presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/PushLimitThroughProject.java

```diff
@@ -32,7 +32,11 @@
     private static final Capture<ProjectNode> CHILD = newCapture();
 
     private static final Pattern<LimitNode> PATTERN = limit()
-            .with(source().matching(project().capturedAs(CHILD)));
+            .with(source().matching(
+                    project()
+                            // do not push limit through identity projection which could be there for column pruning purposes
```

Comment:
limit is about cardinality, why is it affected by column pruning (which sounds like independent relation property)?

Reply:
Once I read the subsequent commit I understand this. Maybe these two commits should be merged together, then you could reference the other rule in the comment here?